### PR TITLE
Update docs to include timeouts

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #486

### CODE REVIEW

File: website/docs/plugin/framework/data-sources/timeouts.mdx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+```go
+type exampleDataSourceData struct {
+    /* ... */
+    Timeouts    types.Object `tfsdk:"timeouts"`
+```
+
+## Accessing Timeouts in CRUD Functions
+
+Once the model has been populated with the config, state or plan the duration of the timeout can be accessed by calling
+the appropriate helper function and then used to configure timeout behaviour, for instance:
```
Comment (bflad) [inline website/docs/plugin/framework/data-sources/timeouts.mdx:66]:
```suggestion
## Accessing Timeout in Read Method

Call the [`timeouts.Read()` function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-timeouts/timeouts#Read).
```

Similarly in the migration guide. 👍

File: website/docs/plugin/framework/migrating/data-sources/timeouts.mdx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        Blocks: map[string]tfsdk.Block{
+            "timeouts": timeouts.Block(ctx, timeouts.Opts{
+                Read: true,
+            }),
+        },
+```
+
+### Nested Attributes
+
+Where possible, we recommend switching to nested attribute syntax during migration.
```
Comment (bflad) [inline website/docs/plugin/framework/migrating/data-sources/timeouts.mdx:44]:
```suggestion
If a practitioner breaking change is acceptable during migration, we recommend switching to nested attribute syntax instead.
```

**Review by bflad — COMMENTED** [review]
Just a few little things, then should be good to go 👍

File: website/docs/plugin/framework/data-sources/timeouts.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+The reality of cloud infrastructure is that it typically takes time to perform operations such as booting operating systems, discovering services, and replicating state across network edges. As the provider developer you should take known delays in data source APIs into account in the `Read` function of the data source. Terraform supports configurable timeouts to assist in these situations.
+
+The Framework can be used in conjunction with the [terraform-plugin-framework-timeouts](https://github.com/hashicorp/terraform-plugin-framework-timeouts) module in order to allow defining timeouts in configuration and have them be available in the `Read` function.
+
+## Specifying Timeouts in Configuration
+
+Timeouts can be specified using nested attributes if [protocol version 6](https://www.terraform.io/plugin/how-terraform-works#protocol-version-6) is being used.
+
+Refer to [nested block](../migrating/data-sources/timeouts#nested-block) if you are migrating timeouts from SDKv2 to the Framework.
```
[the remark this answers is not in the document]
Comment (bendbennett) [reply website/docs/plugin/framework/data-sources/timeouts.mdx:17]: Have added the same information to the resources/timeouts.mdx file too

**Review by bflad — APPROVED** [review]
Looks good to me 🚀

**Pull request merged**